src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,84 @@
+package duke.storage;
+
+import java.util.ArrayList;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
+import java.io.File;
+
+import duke.dukeexception.DukeSaveFileCorruptedError;
+import duke.task.Task;
+import duke.parser.Parser;
+
+public class Storage {
+
+    private String saveFilePath;
+
+    public Storage(String saveFilePath) {
+        this.saveFilePath = saveFilePath;
+    }
+
+    public ArrayList<Task> load() throws DukeSaveFileCorruptedError {
+        try {
+            FileReader tasksFileReader = new FileReader(new File(saveFilePath));
+            BufferedReader bTasksFileReader = new BufferedReader(tasksFileReader);
+            return readTasksFile(bTasksFileReader);
+        } catch (FileNotFoundException e) {
+            createNewSaveFile();
+        } catch (DukeSaveFileCorruptedError e) {
+            createNewSaveFile();
+        }
+        return new ArrayList<Task>();
+    }
+
+    public void saveTasks(ArrayList<Task> tasks) {
+        try {
+            new File(saveFilePath).delete();
```

Comment:
You can rewrite the file instead of deleting and creating the file every time. The File class can check if file exists. If it fails to read or the file is missing, then you can create the file.